controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -75,6 +76,7 @@
     private final RESTAuthHelper restAuthHelper;
     private final LocalController localController;
     private final ConnectionFactory connectionFactory;
+    private final AuthorizationResource authorizationResource = new AuthorizationResourceImpl();
```

Comment:
Just curious where the toggle is to use the LegacyAuthorizationImpl() ?

Might be useful to implementors of the auth handler interface could take the new Pravega code and have a way to stick to the old format while they're updating their implementation to handle the new format ( else they'll break right away once Pravega starts asking about "prn://scope:scope1" and they really only understand  "/scope1" right ?).

I saw there IS a purposely implemented LegacyAuthorizationResourceImpl class but other than in tests, I didn't see it getting used (there are a lot of files in this PR though so I might have missed a subtlety somewhere :))

Reply:
No, you are right - there is no toggle built for the `LegacyAuthorizationImpl`. I've kept the code for ease of comparison between the new and old formats, as we transition from the old to the new format. 

With this code, the resource strings passed to the Auth Handler implementation will all be in the new format. Also, the built-in Password Auth Handler has been modified to take in ACLs containing patterns that match the new format. IMO, supporting both old and new formats would make the code a bit too complex (I had attempted it already) and cause problems in upgrade scenarios, without adding enough value to justify it. 

What do you say?

Reply:
I understand.  Cost vs reward.  It just means that implementors of the AuthHandler interface can't upgrade to Pravega 0.9 until they've updated their plugin code.  That said the changes imposed by the new format are not a big deal.  Also the auth handler implementation doesn't have to change the way IT stores its version of acls.  It simply has to adapt to the new way Pravega will "ask about" the resources being authorized and adjust its mapping. ie. "/scope1" is now "prn::/scope:scope1"

There are probably not many of those implementations out there at this time so that's probably ok.    

I can see though that in the password file impl, you've revamped also the format of the acls themselves since they follow closely the resource format that Pravega itself uses.

One thing I saw is that you have a LegacyAclAuthorizer; that is so somehow one can still use their old password file, correct?    The new PasswodFileHandler will still deal with new resources name being queried on, but it'll be able to map those to old and new ACLs.    Is there a setting/path in the code on the controller side that specifically allows setting that "oldFormat" flag? https://github.com/pravega/pravega/blob/a000a66776c32c3e475cfd7671eca45a1fbb5b69/controller/src/main/java/io/pravega/controller/server/security/auth/handler/impl/PasswordAuthHandler.java#L52

The documentation for the format of the acls in the password file will need to be updated (I'm thinking in particular the pravega cli in the tools repo that produces that password file)